tests/baselines/reference/derivedClassSuperProperties.errors.txt

```diff
@@ -0,0 +1,606 @@
+    }
+    
+    class DerivedWithObjectAccessorsUsingThisInBodies extends Base {
+        propName = "prop";
+        constructor() {
+            const obj = {
+                _prop: "prop",
+                get prop() {
+                    return this._prop;
+                },
+                set prop(param) {
+                    this._prop = param;
+                }
+            };
+            super();
+        }
+    }
+    
+    class DerivedWithObjectComputedPropertyBody extends Base {
+        propName = "prop";
+        constructor() {
+        ~~~~~~~~~~~~~~~
+            const obj = {
+    ~~~~~~~~~~~~~~~~~~~~~
+                prop: this.propName,
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                      ~~~~
+!!! error TS17009: 'super' must be called before accessing 'this' in the constructor of a derived class.
+            };
+    ~~~~~~~~~~
+            super();
+    ~~~~~~~~~~~~~~~~
+        }
+    ~~~~~
+!!! error TS2376: A 'super' call must be the first statement in the constructor to refer to 'super' or 'this' when a derived class contains initialized properties, parameter properties, or private identifiers.
+    }
+    
+    class DerivedWithObjectComputedPropertyName extends Base {
+        propName = "prop";
+        constructor() {
+        ~~~~~~~~~~~~~~~
+            const obj = {
+    ~~~~~~~~~~~~~~~~~~~~~
+                [this.propName]: true,
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                 ~~~~
+!!! error TS17009: 'super' must be called before accessing 'this' in the constructor of a derived class.
+            };
+    ~~~~~~~~~~
+            super();
+    ~~~~~~~~~~~~~~~~
+        }
+    ~~~~~
+!!! error TS2376: A 'super' call must be the first statement in the constructor to refer to 'super' or 'this' when a derived class contains initialized properties, parameter properties, or private identifiers.
+    }
+    
+    class DerivedWithObjectMethod extends Base {
+        prop = true;
+        constructor() {
+            const obj = {
+                getProp() {
+                    return this;
+                },
+            };
+            super();
+        }
+    }
+    
+    let a, b;
+    
+    const DerivedWithLoops = [
+        class extends Base {
+            prop = true;
+            constructor() {
+                for(super();;) {}
+                    ~~~~~~~
+!!! error TS2798: A 'super' call must be a root-level statement within a constructor of a derived class that contains initialized properties, parameter properties, or private identifiers.
+            }
+        },
+        class extends Base {
+            prop = true;
+            constructor() {
+                for(a; super();) {}
+                       ~~~~~~~
+!!! error TS1345: An expression of type 'void' cannot be tested for truthiness.

```

Comment:
This error is interesting because the result of `super()` is certainly not `void`. This isn't something you need to address, however.